### PR TITLE
Bumping Supported Ruby to 2.4.x

### PR DESCRIPTION
**Pull request by cassiascheffer**
This PR bumps the required Ruby Version to >= 2.4.x

### CODE REVIEW

**Review by erawk — APPROVED** [review]

**Comment by cassiascheffer** [comment]
@erawk Do you think this is too restrictive for others who might use this gem? I could bump to a 2.4 version instead.

**Comment by erawk** [comment]
> @erawk Do you think this is too restrictive for others who might use this gem? I could bump to a 2.4 version instead.

You could keep to the `2.4.x` and run CI tests against latest stable `2.4.x`, `2.5.x`, `2.6.x` rubies too. This could identify incompatible versions of rubies for users of this client early.

**Comment by cassiascheffer** [comment]
@erawk I've updated thi to test against three different rubies.

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+      - run: |
+          gem install bundler
+          bundle config path vendor/bundle
+          bundle install --jobs 4 --retry 3
+      - name: Run Tests
+        run: |
+          bundle exec rake spec
```
Comment (harrylewis) [inline .github/workflows/ruby.yml:34]: We could add linting here too soon, I think most of the code would pass.

**Review by erawk — APPROVED** [review]
👋 CircleCI!

👏 For 💎 💎 💎 !

File: .github/workflows/ruby.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby: [ '2.4.x', '2.5.x', '2.6.x']
```
Comment (harrylewis) [inline .github/workflows/ruby.yml:11]: Awesome to have this tested against future versions!

**Review by harrylewis — APPROVED** [review]
Great to become more modern, and also to future-proof our work with testing against other rubies.

**Pull request merged**